src/erc7730/common/abi.py

```diff
@@ -39,14 +41,12 @@ def compute_signature(abi: Function) -> str:
     return abi_to_signature(abi_function)
 
 
-def reduce_signature(signature: str) -> str | None:
-    """Remove parameter names from a function signature. return None if the signature is invalid."""
-    if match := _SOLIDITY_FUNCTION_RE.fullmatch(signature):
-        function_name = match.group(1)
-        parameters = match.group(2).split(",")
-        parameters_types = [param.strip().partition(" ")[0] for param in parameters]
-        return f"{function_name}({','.join(parameters_types)})"
-    return None
+def reduce_signature(signature: str) -> str:
+    """Remove parameter names and spaces from a function signature. Behaviour is undefined on invalid signature."""
+    # Note: Implementation is hackish, but as parameter types can be tuples that can be nested,
```

Comment:
It's maybe easier to move to using a proper parser rather than regexes, like I did for paths, it's quite easy to implement

Reply:
I've thought about using lark but stopped without trying because of: https://lark-parser.readthedocs.io/en/stable/grammar.html#terminals
> Terminals are a linear construct, and therefore may not contain themselves (recursion isn’t allowed).

As types can be tuples of tuples, I thought this could not be easily implemented with lark but I'm not 100% sure this is the kind of recursion that isn't allowed

Reply:
ok rules can be recursive, that makes way more sense.
I'll merge this PR as it will make the CI green but I'll work on a real parser immediately